wallet/src/de/schildbach/wallet/ui/BackupWalletDialogFragment.java

```diff
@@ -232,7 +248,39 @@ private void updateView() {
         positiveButton.setEnabled(hasPassword && hasPasswordAgain);
     }
 
-    private void backupWallet(final String password) {
+    private void backupWallet(final String backupPassword) {
+        try {
+            final Protos.Wallet walletProto = new WalletProtobufSerializer().walletToProto(wallet);
+
+            SecurityGuard securityGuard = new SecurityGuard();
+            if (wallet.isEncrypted()) {
+                String walletPassword = securityGuard.retrievePassword();
+                backgroundThread = new HandlerThread("backgroundThread", Process.THREAD_PRIORITY_BACKGROUND);
+                backgroundThread.start();
+                backgroundHandler = new Handler(backgroundThread.getLooper());
```

Comment:
This could (or rather should) be moved to `BackupWalletDialogFragment.onCreate(...)` as a one-time operation.

Reply:
fixed here:  https://github.com/dashevo/dash-wallet/pull/500/commits/e03147e597783a6ba624e5c53b9d6b44be3e4c0e